src/plugins/vis_type_timeseries/public/application/components/aggs/agg_select.tsx

```diff
@@ -224,29 +227,40 @@ const specialAggs = [
 
 const allAggOptions = [...metricAggs, ...pipelineAggs, ...siblingAggs, ...specialAggs];
 
-function filterByPanelType(panelType) {
-  return (agg) => {
+function filterByPanelType(panelType: string) {
+  return (agg: AggSelectOption) => {
     if (panelType === 'table') return agg.value !== 'series_agg';
     return true;
   };
 }
 
-function AggSelectUi(props) {
+interface AggSelectUiProps {
+  id: string;
+  panelType: string;
+  siblings: MetricsItemsSchema[];
+  value: string;
+  uiRestrictions?: UIRestrictions;
+  onChange: (currentlySelectedOptions: AggSelectOption[]) => void;
+}
+
+export function AggSelect(props: AggSelectUiProps) {
   const { siblings, panelType, value, onChange, uiRestrictions, ...rest } = props;
 
   const selectedOptions = allAggOptions.filter((option) => {
     return value === option.value && isMetricEnabled(option.value, uiRestrictions);
   });
 
-  let enablePipelines = siblings.some((s) => !!metricAggs.find((m) => m.value === s.type));
+  let enablePipelines = siblings.some(
+    (s: { id: string; type: string }) => !!metricAggs.find((m) => m.value === s.type)
```

Comment:
Please don't add extra types like e.g.` { id: string; type: string }`. You already set type for `siblings` variable.